src/components/Channel/Channel.tsx

```diff
@@ -737,7 +737,7 @@ const ChannelWithContext = <
       client.off('connection.changed', connectionChangedHandler);
       channel?.off(handleEvent);
     };
-  }, [channelId]);
+  }, [channelId, connectionRecoveredHandler, handleEvent]);
```

Comment:
I know you probably tested/are aware but we sure we want these 2 functions passed in here if they're not useCallback'd?